File: Python/OOP/Inheritance/Multi-level inheritance (example 3).py
class Institute:
    __institute_name = "Institute of Engineering"  # just a random name
    __eiin = 654321

    def instituteInfo(self):
        print("Institute info: \n--------------")
        print("Name:", self.__institute_name)
        print("EIIN:", self.__eiin)


class Teacher(Institute):
    teacher_name = "Guido van Rossum"
    subject = "Python Programming"

    def teacherInfo(self):
        print("\nTeacher info: \n--------------")
        print("Name   :", self.teacher_name)
        print("Subject:", self.subject)


class Student(Teacher):
    student_name = "Abdullah Al Naiem"
    id = 123456

    def studentInfo(self):
        print("\nStudent info: \n--------------")
        print("Name:", self.student_name)
        print("ID  :", self.id)
        print("Class of:", self.subject)


class multiLevelInheritance:
    get = Student()
    get.instituteInfo()
    get.teacherInfo()
    get.studentInfo()

File: Java/OOP/Inheritance/Multi-level inheritance (example 3).java
class Institute {
    private String institute_name = "Institute of Engineering"; // just a random name
    private int eiin = 654321;

    void instituteInfo() {
        System.out.println("Institute info: \n--------------");
        System.out.println("Name: " + institute_name);
        System.out.println("EIIN: " + eiin);
    }
}

class Teacher extends Institute {
    String teacher_name = "James Gosling";
    String subject = "Java Programming";

    void teacherInfo() {
        System.out.println("\nTeacher info: \n--------------");
        System.out.println("Name   : " + teacher_name);
        System.out.println("Subject: " + subject);
    }
}

class Student extends Teacher {
    String student_name = "Abdullah Al Naiem";
    int id = 123456;

    void studentInfo() {
        System.out.println("\nStudent info: \n--------------");
        System.out.println("Name: " + student_name);
        System.out.println("ID  : " + id);
        System.out.println("Class of: " + subject);
    }
}

class multiLevelInhertance {
    public static void main(String[] args) {
        Student get = new Student();
        get.instituteInfo();
        get.teacherInfo();
        get.studentInfo();
    }
}